torchaudio/transforms.py

```diff
@@ -1730,242 +1703,100 @@ def __init__(
-
-        Args:
-            psd_s (torch.tensor): covariance matrix of speech
-                Tensor of dimension `(..., freq, channel, channel)`
-            psd_n (torch.Tensor): covariance matrix of noise
-                Tensor of dimension `(..., freq, channel, channel)`
-            reference_vector (torch.Tensor): one-hot reference channel matrix
-
-        Returns:
-            torch.Tensor: the enhanced STFT
-                Tensor of dimension `(..., freq, channel, 1)`
-        """
-        phi = torch.linalg.solve(psd_n, psd_s)  # psd_n.inv() @ psd_s
-        stv = torch.einsum("...fec,...c->...fe", [phi, reference_vector])
-        stv = stv.unsqueeze(-1)
-        stv = torch.matmul(phi, stv)
-        stv = torch.matmul(psd_s, stv)
-        return stv
-
-    def _apply_beamforming_vector(self, specgram: torch.Tensor, beamform_vector: torch.Tensor) -> torch.Tensor:
-        r"""Apply the beamforming weight to the noisy STFT
-        Args:
-            specgram (torch.tensor): multi-channel noisy STFT
-                Tensor of dimension `(..., channel, freq, time)`
-            beamform_vector (torch.Tensor): beamforming weight matrix
-                Tensor of dimension `(..., freq, channel)`
-
-        Returns:
-            torch.Tensor: the enhanced STFT
-                Tensor of dimension `(..., freq, time)`
-        """
-        # (..., channel) x (..., channel, freq, time) -> (..., freq, time)
-        specgram_enhanced = torch.einsum("...fc,...cft->...ft", [beamform_vector.conj(), specgram])
-        return specgram_enhanced
-
-    def _tik_reg(self, mat: torch.Tensor, reg: float = 1e-7, eps: float = 1e-8) -> torch.Tensor:
-        """Perform Tikhonov regularization (only modifying real part).
-        Args:
-            mat (torch.Tensor): input matrix (..., channel, channel)
-            reg (float, optional): regularization factor (Default: 1e-8)
-            eps (float, optional): a value to avoid the correlation matrix is all-zero (Default: 1e-8)
-
-        Returns:
-            torch.Tensor: regularized matrix (..., channel, channel)
-        """
-        # Add eps
-        C = mat.size(-1)
-        eye = torch.eye(C, dtype=mat.dtype, device=mat.device)
-        with torch.no_grad():
-            epsilon = _get_mat_trace(mat).real[..., None, None] * reg
-            # in case that correlation_matrix is all-zero
-            epsilon = epsilon + eps
-        mat = mat + epsilon * eye[..., :, :]
-        return mat
-
-    def forward(
-        self, specgram: torch.Tensor, mask_s: torch.Tensor, mask_n: Optional[torch.Tensor] = None
-    ) -> torch.Tensor:
+    def forward(self, specgram: Tensor, mask_s: Tensor, mask_n: Optional[Tensor] = None) -> Tensor:
         """Perform MVDR beamforming.
 
         Args:
-            specgram (torch.Tensor): the multi-channel STF of the noisy speech.
+            specgram (Tensor): the multi-channel STF of the noisy speech.
```

Comment:
Elsewhere I think you refer  to STF as STFT?  And perhaps you should specifiy that it is complex? 
(I assume it is..)

Reply:
Oops, it's a typo, should be STFT. Maybe calling it `spectrogram` is better for such description.

Reply:
https://en.wikipedia.org/wiki/Spectrogram :

> A spectrogram is a visual representation of the [spectrum](https://en.wikipedia.org/wiki/Spectral_density) of [frequencies](https://en.wikipedia.org/wiki/Frequencies) of a signal as it varies with time. When applied to an [audio signal](https://en.wikipedia.org/wiki/Audio_signal), spectrograms are sometimes called sonographs, voiceprints, or voicegrams. When the data are represented in a 3D plot they may be called [waterfall displays](https://en.wikipedia.org/wiki/Waterfall_display).

For me, the `spectrogram` is a real valued signal, e.g. the magnitude or the power of the STFT signal.
According to wikipedia, it is the visualization of a signal, with a time and a frequency axis.
If you want to keep the name, I think it is important to mention, that it is complex values,
because often this name is connected with something real values.

Reply:
This is true, but I do see some papers use `spectrum` to refer the complex valued STFT, like the phase-sensitive mask paper. Since torchaudio support `Spectrogram` module that returns complex Tensor, using the same term makes it more consistent. But yeah, definitely mention the `complex-valued` helps users understand better.

Reply:
So `spectrum` would maybe a better name?
For that, it is at least difficult to find a definition, that it is real or complex valued.

Reply:
> So spectrum would maybe a better name?

Sure, let's use it in the docstring.

Reply:
Addressed in https://github.com/pytorch/audio/pull/2181/commits/4c1a539aefaa9bb27e2f3628f694cf107386001d